### PR TITLE
Cody VSCode: Add support for verified email check message when connected to app; and improve connected status message

### PR DESCRIPTION
**Pull request by marekweb**
### Show verified email message when connecting Cody extension to App

When connecting the Cody extension to App, it's expected to act similarly to dotcom in that requires an email address and if the user doesn't have a verified email address, it should show the same message.

This PR checks if we are connecting to App, and if so it will treat the email verification logic the same as if we were connecting to sourcegraph.com, so we also get the same error message for an unverified email.

Background info: App will basically proxy the request to dotcom so the response to `currentUser { id, hasVerifiedEmail }` will be similar to a dotcom response. The only difference, then, is that Cody extension holds an access token to App and App holds an access token to dotcom.

### Video

https://github.com/sourcegraph/sourcegraph/assets/602886/2d65e1f4-b8c8-4790-a476-b41fdfccb1ca

### Update "Connected to" display

This is a cosmetic change to show "Connected to Sourcegraph App" instead of "Connected to http://localhost:3080":

Screenshot 2023-05-17 at 10 58 04 AM

## Test plan

-  Connect App to a dotcom account 
-  Test Cody extension with an unverified email address in the dotcom account
-  Test Cody extension after verifying the email address in the dotcom account

### CODE REVIEW

File: client/cody/src/main.ts [hunk 1]
```diff
@@ -192,7 +192,7 @@ const register = async (
                 if (token && token.length > 8) {
                     await secretStorage.store(CODY_ACCESS_TOKEN_SECRET, token)
                     const authStatus = await getAuthStatus({
-                        serverEndpoint: DOTCOM_URL.href,
+                        serverEndpoint,
```
Comment (marekweb) [inline client/cody/src/main.ts:195]: This fixes a regression; serverEndpoint is already set above to DOTCOM_URL if we're connecting to dotcom, or it's set to LOCAL_APP_URL if we're connecting to App so we don't want to overwrite it here.

**Comment by abeatrix** [comment]
Is there a setting i'd need to change to test? I have the old feature flag set and app opened, but I don't see the button to login with app:
image

**Review by abeatrix — APPROVED** [review]
I don't have the release build of App to test the button but the code looks good to me

**Review by philipp-spiess — APPROVED** [review]
Sweet!

**Review by slimsag — APPROVED** [review]

**Review by limitedmage — COMMENTED** [review]
Can we change the error message to "Sourcegraph.com account"? (instead of instance) 
image

**Comment by mrnugget** [comment]
@limitedmage yup, makes sense: https://github.com/sourcegraph/sourcegraph/pull/52131

**Pull request merged**